adal/src/main/java/com/microsoft/aad/adal/BrokerProxy.java

```diff
@@ -159,15 +159,23 @@ public boolean canUseLocalCache() {
      * App needs to give permission to AccountManager to use broker.
      */
     private boolean verifyManifestPermissions() {
-        PackageManager pm = mContext.getPackageManager();
-        boolean permission = PackageManager.PERMISSION_GRANTED == pm.checkPermission(
+        final PackageManager pm = mContext.getPackageManager();
+        final boolean permission;
+        
+        if (Build.VERSION.SDK_INT < Build.VERSION_CODES.M) {
+            permission = PackageManager.PERMISSION_GRANTED == pm.checkPermission(
                 "android.permission.GET_ACCOUNTS", mContext.getPackageName())
                 && PackageManager.PERMISSION_GRANTED == pm.checkPermission(
                         "android.permission.MANAGE_ACCOUNTS", mContext.getPackageName())
                 && PackageManager.PERMISSION_GRANTED == pm.checkPermission(
                         "android.permission.USE_CREDENTIALS", mContext.getPackageName());
+        } else {
+            permission = PackageManager.PERMISSION_GRANTED == pm.checkPermission(
+                "android.permission.GET_ACCOUNTS", mContext.getPackageName());
```

Comment:
This check should be combined as Ashish said. 
You can do in this way:
if (Build.Version.SDK_INT < M) {
    // check manage_account and use_credential
}

// check get_account